emop-maven-plugin/src/main/java/edu/cornell/AffectedSpecsMojo.java

```diff
@@ -34,11 +49,76 @@ public void execute() throws MojoExecutionException {
         MessageHandler mh = new MessageHandler();
         compiler.run(arguments, mh);
         IMessage[] ms = mh.getMessages(IMessage.WEAVEINFO, false);
-        Writer.writeToFile(Arrays.asList(ms), getArtifactsDir() + File.separator + "join-points");
+        computeMapFromMessage(ms);
+        writeMapToFile(OutputFormat.TXT);
         getLog().info("[eMOP] Number of impacted classes: " + getImpacted().size());
         getLog().info("[eMOP] Number of messages to process: " + Arrays.asList(ms).size());
     }
 
+    /**
+     * Compute a mapping from affected classes to specifications based on the messages from AJC.
+     * TODO: Handling output messages are definitely slower than reading them directly from AJC
+     * @param ms An array of IMessage objects
+     */
+    private void computeMapFromMessage(IMessage[] ms) throws MojoExecutionException {
+        for (IMessage message : ms) {
+            String[] lexedMessage = message.getMessage().split("'");
+            String key = lexedMessage[CLASS_INDEX_IN_MSG];
+            String value = lexedMessage[SPEC_INDEX_IN_MSG].substring(TRIMMED_SPEC_NAME_INDEX);
+            if (!classToSpecs.containsKey(key)) {
+                classToSpecs.put(key, new HashSet<>());
+            }
+            classToSpecs.get(key).add(value);
+        }
+    }
+
+    /**
+     * Write map from class to specs in either text or binary format.
+     * @param format Output format of the map, text or binary
+     */
+    private void writeMapToFile(OutputFormat format) throws MojoExecutionException {
+        switch (format) {
+            case BIN:
+                // Referenced from https://www.geeksforgeeks.org/how-to-serialize-hashmap-in-java/
+                try (FileOutputStream fos
+                             = new FileOutputStream(getArtifactsDir() + File.separator + "classToSpecs.bin");
+                     ObjectOutputStream oos = new ObjectOutputStream(fos)) {
+                    oos.writeObject(classToSpecs);
+                } catch (IOException ex) {
+                    ex.printStackTrace();
+                }
+                break;
+            case TXT:
+                try (FileWriter writer
+                             = new FileWriter(getArtifactsDir() + File.separator + "classToSpecs.txt")) {
+                    for (Map.Entry<String, Set<String>> entry : classToSpecs.entrySet()) {
+                        writer.write(entry.getKey() + ":" + String.join(",", entry.getValue()) + "\n");
```

Comment:
If you use `writeln` instead of `write`, you will not need `+ "\n"` at the end

Reply:
Use PrintWriter instead of FileWriter; the former has a [writeln method](https://docs.oracle.com/javase/8/docs/api/java/io/PrintWriter.html#println-java.lang.String-)